modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1234,42 +1232,47 @@ private CompletableFuture<Void> dropTableAsyncInternal(String name) {
             if (tbl == null) {
                 dropTblFut.completeExceptionally(new TableNotFoundException(name));
             } else {
-                tablesCfg.tables()
+                List<CompletableFuture<Void>> idxsChange = new ArrayList<>();
+
+                CompletableFuture<Void> tblOps = tablesCfg.tables()
                         .change(change -> {
                             TableView tableCfg = change.get(name);
 
                             if (tableCfg == null) {
                                 throw new TableNotFoundException(name);
                             }
 
-                            NamedListView<TableIndexView> idxView = tablesCfg.indexes().value();
+                            change.delete(name);
+
+                            List<String> indicesNames = tablesCfg.indexes().value().namedListKeys();
 
-                            boolean idxFound = idxView.namedListKeys().stream()
-                                    .anyMatch(idx -> idxView.get(idx).tableId().equals(tbl.tableId()));
+                            List<String> indexes = indicesNames.stream().filter(idx ->
+                                    tablesCfg.indexes().get(idx).tableId().value().equals(tbl.tableId())).collect(Collectors.toList());
 
-                            //TODO: https://issues.apache.org/jira/browse/IGNITE-17562
-                            // Let's drop orphaned indices instantly.
-                            if (idxFound) {
-                                throw new IgniteException("Can't drop table with indices.");
+                            for (String indexName : indexes) {
+                                idxsChange.add(tablesCfg.indexes().change(chg -> chg.delete(indexName)));
```

Comment:
What if external changer will be called more than once?
Should `idxsChange` be cleaned at first?

Reply:
Does it make sense to move the loop into the changer?
`tablesCfg.indexes().change(chg ->indexes.stream().foreach(indexName -> chg.delete(indexName)))`